[git-webkit] Make git-webkit revert a two step workflow

facetothefate opened the pull request:
#### 08be631a9ce03f26ef62bb2d2f7b18423d4d8bff
<pre>
[git-webkit] Make git-webkit revert a two step workflow
<a href="https://bugs.webkit.org/show_bug.cgi?id=240192">https://bugs.webkit.org/show_bug.cgi?id=240192</a>
rdar://92875608

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.create_pull_request):
(PullRequest.add_comment_to_reverted_commit_bug_tracker):
(PullRequest.is_revert_commit):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
(Revert):
(Revert.parser):
(Revert.revert_commit):
(Revert.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/revert_unittest.py:
(TestRevert.test_github):
(test_update):

Canonical link: <a href="https://commits.webkit.org/250670@main">https://commits.webkit.org/250670@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294376">https://svn.webkit.org/repository/webkit/trunk@294376</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
```diff
@@ -110,16 +134,24 @@ def add_comment_to_reverted_commit_bug_tracker(cls, repository, args):
         return 0
 
+    @classmethod
+    def is_revert_commit(cls, commit):
```
I think this function should belong to the commit object.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
```diff
@@ -100,7 +124,7 @@ def revert_commit(cls, args, repository, **kwargs):
 
     @classmethod
-    def add_comment_to_reverted_commit_bug_tracker(cls, repository, args):
+    def add_comment_to_reverted_commit_bug_tracker(cls, repository, args, commit):
```
We could have this owned by the `PullRequest` object, but I agree this function makes more sense attached to the `Revert` class.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
```diff
@@ -354,6 +354,10 @@ def create_pull_request(cls, repository, args, branch_point):
+            # avoid circular import
+            from .revert import Revert
+            if Revert.is_revert_commit(commits[0]):
+                Revert.add_comment_to_reverted_commit_bug_tracker(repository, args, commits[0])
```
This will result in a double comment, right? Because we are still making the normal comment on line 370. That also makes me wonder if the `PullRequest` program should actually own the `add_comment_to_reverted_commit_bug_tracker` since we already own the re-opening and standard comment.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
```diff
@@ -354,6 +383,8 @@ def create_pull_request(cls, repository, args, branch_point):
                 return 1
             print("Created '{}'!".format(pr))
+            if cls.is_revert_commit(commits[0]):
+                cls.add_comment_to_reverted_commit_bug_tracker(repository, args, commits[0])
```
We should pass the `pr` object into this function call, `PullRequest.find_existing_pull_request` will make an extra network request otherwise. We could even make it an optional argument and use the network call if our caller didn't specify the object

litherum committed:
[WebGPU] Work around https://bugs.webkit.org/show_bug.cgi?id=240219
https://bugs.webkit.org/show_bug.cgi?id=240441

https://bugs.webkit.org/show_bug.cgi?id=240219 is about a bug where we can't
say "optional thingy = 0" in an IDL file. This patch works around that in WebGPU
by replacing it with "optional thingy" and then "thingy.value_or(0)" in the C++
code. This is temporary, until https://bugs.webkit.org/show_bug.cgi?id=240219 is
fixed.

No new tests because there is no behavior change.

* Modules/WebGPU/GPUBuffer.cpp:
(WebCore::GPUBuffer::mapAsync):
(WebCore::GPUBuffer::getMappedRange):
* Modules/WebGPU/GPUBuffer.h:
* Modules/WebGPU/GPUBuffer.idl:

Canonical link: https://commits.webkit.org/250668@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294373 268f45cc-cd09-0410-ab3c-d52691b4dbfc

yury-s committed:
Web Inspector: support EventSource resource type in Network Panel
https://bugs.webkit.org/show_bug.cgi?id=240326

Reviewed by Devin Rousso.

* http/tests/inspector/network/eventsource-type-expected.txt: Added.
* http/tests/inspector/network/eventsource-type.html: Added.
* http/tests/inspector/network/resources/event-source.py: Added.
Web Inspector: support EventSource resource type in Network Panel
https://bugs.webkit.org/show_bug.cgi?id=240326

* inspector/protocol/Page.json: added EventSource resource type.
Web Inspector: support EventSource resource type in Network Panel
https://bugs.webkit.org/show_bug.cgi?id=240326

Network requests initiated by EventSource now have their own requester
type which allows to distinguish them from other requests.

Test: http/tests/inspector/network/eventsource-type.html

* inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::resourceTypeJSON):
(WebCore::InspectorPageAgent::inspectorResourceType):
* inspector/agents/InspectorPageAgent.h:
* page/EventSource.cpp:
(WebCore::EventSource::connect):
* platform/network/ResourceRequestBase.h:
Web Inspector: support EventSource resource type in Network Panel
https://bugs.webkit.org/show_bug.cgi?id=240326

EventSource initiated requests are now displayed with their own type
in the network panel.

* Localizations/en.lproj/localizedStrings.js:
* UserInterface/Controllers/NetworkManager.js:
(WI.NetworkManager.prototype.canBeOverridden):
* UserInterface/Models/Resource.js:
(WI.Resource.displayNameForType):
* UserInterface/Views/NetworkTableContentView.js:
(WI.NetworkTableContentView.displayNameForResource):

Canonical link: https://commits.webkit.org/250669@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294374 268f45cc-cd09-0410-ab3c-d52691b4dbfc

facetothefate committed:
[git-webkit] Make git-webkit revert a two step workflow
https://bugs.webkit.org/show_bug.cgi?id=240192
rdar://92875608

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.create_pull_request):
(PullRequest.add_comment_to_reverted_commit_bug_tracker):
(PullRequest.is_revert_commit):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
(Revert):
(Revert.parser):
(Revert.revert_commit):
(Revert.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/revert_unittest.py:
(TestRevert.test_github):
(test_update):

Canonical link: https://commits.webkit.org/250670@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294376 268f45cc-cd09-0410-ab3c-d52691b4dbfc
